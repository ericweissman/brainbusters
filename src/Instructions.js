import React, { Component } from 'react'
class Instructions extends Component {
  render() {
    return (
      <div className="instructions">
        <h5>How It Works</h5>
        <p>Use this app to build you skills with tricky methods you'll encounter in Javascript!</p>
        <p>Each card is a multiple choice question. Click on the answer you think is correct!</p>
        <p>If you get the answer incorrect, you will have the option to save it later for review in your Study List.</p>
        <p>If you want to view the questions you missed, choose Study List Only from the dropdown!</p>
        <button onClick={this.props.hideInstructions}>Got it!</button>
      </div>
    )
  }
}

export default Instructions;