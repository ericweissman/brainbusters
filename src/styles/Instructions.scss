.instructions {
  height: 125px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-right: 50px;
  
  h5 {
    font-family: $title;
    margin: 3px 0;
    font-size: 20px;
    color: $blue;
  }
  p {
    font-size: 12px;
    line-height: 5px;
    color: $white
  }

  button {
    @include got-it-button
  }
}