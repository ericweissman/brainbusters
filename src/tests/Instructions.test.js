import React from 'react';
import Instructions from '../Instructions';
import { shallow } from 'enzyme';

const toggleInstructionsMock = jest.fn()

describe('Instructions', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Instructions 
        toggleInstructions={toggleInstructionsMock}
      />
    )
  });

  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggle instructions on click', () => {
    wrapper.find('button').simulate('click');
    expect(toggleInstructionsMock).toBeCalled()
  });

})