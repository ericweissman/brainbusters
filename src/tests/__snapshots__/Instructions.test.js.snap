// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions should match snapshot when all data is passed correctly 1`] = `
<div
  className="instructions"
>
  <h5>
    How It Works
  </h5>
  <div>
    <p>
      Each card is a multiple choice question. Click on the answer you think is correct!
    </p>
    <p>
      If you get the answer incorrect, it will be added to your Study List to continue practicing.
    </p>
    <p>
      If you want to view the questions you missed, choose Study List Only from the dropdown!
    </p>
  </div>
  <button
    onClick={[MockFunction]}
  >
    Got it!
  </button>
</div>
`;
